pandas/tests/indexing/multiindex/test_partial.py

```diff
@@ -208,6 +215,49 @@ def test_setitem_multiple_partial(self, multiindex_dataframe_random_data):
         expected.loc["bar"] = 0
         tm.assert_series_equal(result, expected)
 
+    def test_partial_getitem_loc_datetime(self):
+        # GH: 25165
+        date_idx = date_range("2019", periods=2, freq="MS")
+        df = DataFrame(
+            list(range(4)),
+            index=MultiIndex.from_product([date_idx, [0, 1]], names=["x", "y"]),
+        )
+        expected = DataFrame(
+            [2, 3],
+            index=MultiIndex.from_product(
+                [[to_datetime("2019-02-01")], [0, 1]], names=["x", "y"]
+            ),
+        )
+        result = df["2019-2":]
+        tm.assert_frame_equal(result, expected)
+        result = df.loc["2019-2":]
+        tm.assert_frame_equal(result, expected)
+
+        result = df.loc(axis=0)["2019-2":]
+        tm.assert_frame_equal(result, expected)
+
+        result = df.loc["2019-2":, :]
+        tm.assert_frame_equal(result, expected)
+
+        df2 = df.swaplevel(0, 1).sort_index()
+        expected = expected.swaplevel(0, 1)
+
+        result = df2.loc[:, "2019-02":, :]
+        tm.assert_frame_equal(result, expected)
```

Comment:
if any way to parameterize this would be great

Reply:
Eval is the only option I can think of, but would want to avoid this. Don't know how to parametrize the changing position of ``:`` can be done otherwise

Reply:
> the changing position of `:` 

do you mean `"2019-02":` vs `:"2019-02"`?  Those are `slice("2019-02", None)` and `slice(None, "2019-02")`, respectively.

Reply:
Does this take the same code path? Then this would work for sure.

Reply:
Not perfect, but way better. Thanks very much